src/System.Private.Xml/src/System/Xml/NameTable.cs

```diff
@@ -263,5 +231,17 @@ private static bool TextEquals(string str1, char[] str2, int str2Start, int str2
             }
             return true;
         }
+
+        private static int ComputeHash32(string key)
+        {
+            ReadOnlySpan<byte> bytes = key.AsReadOnlySpan().AsBytes();
+            return Marvin.ComputeHash32(bytes, Marvin.DefaultSeed);
```

Comment:
Why not change the seed, belt-and-braces?

Reply:
[`DefaultSeed`](https://github.com/dotnet/corefx/blob/6ded3992a29c48ec3c9e641589b9805a1dbc50a3/src/Common/src/System/Marvin.cs#L106-L115) is random per process run

Reply:
@benaadams yes, this is intentional - this matches what we do in string.GetHashCode() - this prevents pre-generating collisions and creating slow parsed document.
@JonHanna, do you mean to do it per NameTable instance? This should be not guessable already and would slow down if you create multiple instances of XmlDocument - we could consult this with some crypto experts but considering we don't do that anywhere else I do not believe this will add us more security.

Reply:
@krwq I was responding to @JonHanna about it already being random :)

Reply:
😄

Reply:
Nice